src/login/login.ts

```diff
@@ -171,82 +179,29 @@ export async function ensureLoadedProfile (context: AuthenticationContext): Prom
 export async function findAppInstances (
   context: AuthenticationContext,
   theClass: NamedNode,
   isPublic?: boolean
 ): Promise<AuthenticationContext> {
-  // console.log('findAppInstances', { context, theClass, isPublic })
-  if (isPublic === undefined) {
-    // Then both public and private
-    // console.log('finding public app instance')
-    await findAppInstances(context, theClass, true)
-    // console.log('finding private app instance')
-    await findAppInstances(context, theClass, false)
-    // console.log('found public & private app instance', context)
-    return context
-  }
-
-  // Loading preferences is more than loading profile
-  try {
-    // console.log('calling logInLoad', isPublic)
-    await (isPublic
-      ? ensureLoadedProfile(context)
-      : ensureLoadedPreferences(context))
-    // console.log('called logInLoad', isPublic)
-  } catch (err) {
-    widgets.complain(context, `loadIndex: login and load problem ${err}`)
-  }
-  // console.log('awaited LogInLoad!', context)
-  const visibility = isPublic ? 'public' : 'private'
-  try {
-    await loadIndex(context, isPublic)
-  } catch (err) {
-    debug.error(err)
-  }
-  const index = context.index as { [key: string]: Array<NamedNode> }
-  const thisIndex = index[visibility]
-  const registrations = thisIndex
-    .map(ix => solidLogicSingleton.store.each(undefined, ns.solid('forClass'), theClass, ix))
-    .reduce((acc, curr) => acc.concat(curr), [])
-  const instances = registrations
-    .map(reg => solidLogicSingleton.store.each(reg as NamedNode, ns.solid('instance')))
-    .reduce((acc, curr) => acc.concat(curr), [])
-  const containers = registrations
-    .map(reg => solidLogicSingleton.store.each(reg as NamedNode, ns.solid('instanceContainer')))
-    .reduce((acc, curr) => acc.concat(curr), [])
-
-  function unique (arr: NamedNode[]): NamedNode[] {
-    return Array.from(new Set(arr))
-  }
-  context.instances = context.instances || []
-  context.instances = unique(context.instances.concat(instances as NamedNode[]))
-
-  context.containers = context.containers || []
-  context.containers = unique(context.containers.concat(containers as NamedNode[]))
-  if (!containers.length) {
-    return context
-  }
-  // If the index gives containers, then look up all things within them
-  try {
-    await solidLogicSingleton.load(containers as NamedNode[])
-  } catch (err) {
-    const e = new Error(`[FAI] Unable to load containers${err}`)
-    debug.log(e) // complain
-    widgets.complain(context, `Error looking for ${utils.label(theClass)}:  ${err}`)
-    // but then ignore it
-    // throw new Error(e)
-  }
-  for (let i = 0; i < containers.length; i++) {
-    const cont = containers[i]
-    context.instances = context.instances.concat(
-      (await solidLogicSingleton.getContainerMembers(cont.value)).map(uri => solidLogicSingleton.store.sym(uri)) // @@ warning: uses strings not NN
-    )
+  let items = await getScopedAppInstances(store, theClass, context.me)
```

Comment:
context.me can be also null but getScopedAppInstances does not accept that option. This code needs some hardening.

Reply:
Did Ts spot that?  Agreed need 
```
let items = context.me ? await getScopedAppInstances(store, theClass, context.me) : []
```